# Sequence classification

## Get taxonomy classifiers
* Any of the following classifiers can be used in classification
* Place the files in the references directory

### SILVA reference classifier
```{}
degap.seqs(fasta=data/references/silva.nr.align)
```

### RDP reference classifier
```{}
wget --no-check-certificate -N https://www.mothur.org/w/images/c/c3/Trainset16_022016.pds.tgz
tar xvzf Trainset16_022016.pds.tgz
mv trainset16_022016.pds/train* data/references/
rm -rf trainset16_022016.pds
rm Trainset16_022016.pds.tgz
```

### Greengenes reference classifier
```{}
wget --no-check-certificate http://www.mothur.org/w/images/6/68/Gg_13_8_99.taxonomy.tgz
tar xvzf Gg_13_8_99.taxonomy.tgz gg_13_8_99.fasta gg_13_8_99.gg.tax
mv gg* data/references/
rm Gg_13_8_99.taxonomy.tgz
```

### Mock classifier
* Used to estimate error rate
```{}
wget --no-check-certificate https://www.mothur.org/MiSeqDevelopmentData/HMP_MOCK.fasta
mv HMP_MOCK.fasta data/references
```

## Assign taxonomy to sequences
```{}
set.dir(input=data/mothur/, output=data/mothur/)
classify.seqs(fasta=qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta, count=qced.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count_table,reference=data/references/silva.nr.ng.fasta, taxonomy=data/references/silva.nr.tax, cutoff=97)
remove.lineage(fasta=current, count=current, taxonomy=current, taxon=Chloroplast-Mitochondria-unknown-Archaea-Eukaryota-Viruses-Virus)
summary.tax(taxonomy=current, count=current)
```

## Estimate sequencing error rate
```{}
set.current(inputdir=data/mothur, outputdir=data/mothur/seqerrorrate)
get.groups(fasta=qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta, count=qced.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table, taxonomy=qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.nr.wang.pick.taxonomy, groups=Mock-Mock2)

seq.error(fasta=qced.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.fasta, count=qced.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.pick.count_table, reference=data/references/HMP_MOCK.fasta, aligned=F)
```
