# Proposed manuscript 1

## iMAP: Integrative Microbiome Analysis Pipeline {-}

## Abstract {-}

**Background:** Microbiome studies present great opportunities for understanding microbial communities in diverse samples and their impact in environment, public health and disease. Several state-of-the-art bioinformatics tools have enabled microbiologists and ecologists identify the microbial communities living in a given sample. However, most of these tools do not integrate exhaustive analytical or visualization procedure in their workflow. The only way we are going to unleash all biological information hidden in highly multidimensional microbiome data is to integrate analytical workflow with in-depth analysis and comparatively extraordinary data visualization. Effective microbiome analytical and visualization workflow that provides better understanding of microbial communities in a sample is also a great opportunity to new insights and new data-driven hypotheses. We have developed a pipeline named iMAP (Integrative Microbiome Analysis Pipeline) to address some of these challenges. The iMAP pipeline streamline functionalities implemented in published tools such as: Mothur, Kraken, and R packages and makes use of Rmarkdown to produce reproducible web-based reports. The pipeline operates on high throughput 16S rRNA data generated by Illumina platform and performs tunable quality filtering before assigning known taxonomic nomenclatures to the representative sequences. Bacterial abundance is then analyzed to measure the diversity within and between samples. Finally, the results are visualized using integrated R functions and summarized using Rmarkdown.  
 
**Results:** The pipeline output data tables which are then converted to data structures suitable for statistical analysis and for generating publication-quality figures. 

**Conclusions:** The iMAP pipeline accelerates the identification of microbial communities that are present in an environmental sample. The rich visuals produced by the pipeline provides different ways of extracting meaning and better understanding of the multidimensional microbiome data and the reproducible web-based reports facilitate quick sharing with collaborators. We anticipate that users will find this tool broadly useful and adaptable to their needs.

<br>

**Keywords:** Microbiome, 16S rRNA classification, Microbial communities, pipeline, bacterial communities.

\newpage

## Background
## Results
## Discussion
## Conclusions
## MethodsÂ 
## List of abbreviations
## Declarations
## Ethics approval and consent to participate
## Consent for publication
## Availability of data and materials
## Competing interests
## Funding
## Authors' contributions
## Acknowledgements
